xskillscore/core/deterministic.py

```diff
@@ -159,8 +180,110 @@ def pearson_r_p_value(a, b, dim, weights=None, skipna=False):
         b,
         weights,
         input_core_dims=[[new_dim], [new_dim], [new_dim]],
-        kwargs={"axis": -1, "skipna": skipna},
-        dask="parallelized",
+        kwargs={'axis': -1, 'skipna': skipna},
+        dask='parallelized',
+        output_dtypes=[float],
+    )
+
+
+def spearman_r(a, b, dim, weights=None, skipna=False):
+    """
+    Spearman's correlation coefficient.
+
+    Parameters
+    ----------
+    a : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+    b : xarray.Dataset or xarray.DataArray
+        Labeled array(s) over which to apply the function.
+    dim : str, list
+        The dimension(s) to apply the correlation along.
+    weights : xarray.Dataset or xarray.DataArray
+        Weights matching dimensions of ``dim`` to apply during the function.
+        If None, an array of ones will be applied (i.e., no weighting).
+    skipna : bool
+        If True, skip NaNs when computing function.
+
+    Returns
+    -------
+    xarray.DataArray or xarray.Dataset
+        Spearman's correlation coefficient.
+
+    See Also
+    --------
+    scipy.stats.spearman_r
+    xarray.apply_ufunc
+    """
```

Comment:
Mind copying the Reference from _spearman_r e.g.

    Reference
    ---------
    https://github.com/scipy/scipy/blob/v1.3.1/scipy/stats/stats.py#L3613-L3764